cmd/bee/cmd/bcrypt.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func (c *command) initHasherCmd() (err error) {
+	cmd := &cobra.Command{
+		Use:   "bcrypt",
+		Short: "Generate or validate a bcrypt hash",
+		Long: `Generate or validate a bcrypt hash
+
+Takes a single plain text argument in order to generate a bcrypt hash.
+If '--check' flag is provided it will validate the first (plain text) argument against
+the second one, which is expected to be a quoted bcrypt hash.`,
+		Example: `
+$> bee bcrypt super$ecret
+$2a$10$eZP5YuhJq2k8DFmj9UJGWOIjDtXu6NcAQMrz7Zj1bgIVBcHA3bU5u
+
+$> bee bcrypt --check super$ecret '$2a$10$eZP5YuhJq2k8DFmj9UJGWOIjDtXu6NcAQMrz7Zj1bgIVBcHA3bU5u'
+OK: password hash matches provided plain text`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 || len(args) > 2 {
+				return cmd.Help()
+			}
+
+			isCheck := c.config.GetBool("check")
+
+			if isCheck {
+				if len(args) != 2 {
+					fmt.Println("Usage:", "bee bcrypt", "--check", "your-plain-text-password", "'password-hash'")
+					return nil
+				}
+
+				err := bcrypt.CompareHashAndPassword([]byte(args[1]), []byte(args[0]))
+				if err != nil {
+					return errors.New("password hash does not match provided plain text")
+				}
+
+				fmt.Println("OK: password hash matches provided plain text")
+				return nil
+			}
+
+			if len(args) != 1 {
+				return cmd.Help()
+			}
+
+			hashed, err := bcrypt.GenerateFromPassword([]byte(args[0]), bcrypt.DefaultCost)
+			if err != nil {
+				return errors.New("failed to generate password hash")
+			}
+
+			fmt.Println(string(hashed))
```

Comment:
Be aware that you're print not just the actual string here, but also a line feed. It would be good to check that it doesn't make some cli checks fail as a result (e.g. if the hashed password is stored in a file then simply `bee bcrypt --check playtextpass | diff hash_pass -` to see that they match)

Reply:
I think it works:
```
$> go run $(pwd)/cmd/bee bcrypt super$ecret > plaintext
$> go run $(pwd)/cmd/bee bcrypt --check super$ecret `cat plaintext` 
OK: password hash matches provided plain text
```
but I will remove the new line just in case

Reply:
they will always pass in this case, because the command generated both outputs (hence both have exactly the same characters, line feed included) :)